client/src/worldgen.rs

```diff
@@ -155,10 +155,89 @@ pub fn voxels(graph: &mut DualGraph, node: NodeId, cube: Vertex) -> VoxelData {
                 }
             }
         }
+
+        // Planting trees on dirt. Trees consist of a block of wood and a block of leaves.
+        // The leaf block is on the opposite face of the wood block as the dirt block.
+        let loc: usize = 2;
+        let voxel_of_interest_index =
+            index(absolute_subchunk_coords(loc, loc, loc, subchunk_offset));
+        let neighbor_data = voxel_neighbors(
+            na::Vector3::new(loc, loc, loc),
+            subchunk_offset,
+            &mut voxels,
+        );
+
+        let num_void_neighbors = neighbor_data
+            .iter()
+            .filter(|n| n.material == Material::Void)
+            .count();
+
+        // Only plant a tree if there is exactly one adjacent dirt block.
+        if num_void_neighbors == 5 {
+            for i in neighbor_data.iter() {
+                if i.material == Material::Dirt {
+                    voxels.data_mut()[voxel_of_interest_index] = Material::Wood;
+                    let leaf_location = index(i.abs_coords_opposing);
+                    voxels.data_mut()[leaf_location] = Material::Leaves;
```

Comment:
How do we know neither of these coordinates are out of bounds?

Reply:
I have written a test to address this.